nemo/collections/asr/parts/mixins/transcription.py

```diff
@@ -0,0 +1,579 @@
+
+        else:
+            raise ValueError(
+                f"Input `audio` is of type {type(audio[0])}. "
+                "Only `str` (path to audio file), `np.ndarray`, and `torch.Tensor` "
+                "are supported as input."
+            )
+
+    def _transcribe_input_tensor_processing(
+        self, audio_tensors: List[Union[np.ndarray, torch.Tensor]], temp_dir: str, trcfg: TranscribeConfig
+    ):
+        # Check if sample rate is set
+        sample_rate = None
+        if hasattr(self, 'cfg') and 'sample_rate' in self.cfg:
+            sample_rate = self.cfg.sample_rate
+        elif hasattr(self, 'sample_rate'):
+            sample_rate = self.sample_rate
+
+        if sample_rate is None:
+            raise RuntimeError(
+                "Provided `audio` data contains numpy or torch tensors, however the class "
+                "does not have `sample_rate` attribute. Please set `sample_rate` attribute to the model explicitly."
+            )
+
+        ds_config = {
+            'audio_tensors': audio_tensors,
+            'batch_size': trcfg.batch_size,
+            'temp_dir': temp_dir,
+            'num_workers': trcfg.num_workers,
+            'channel_selector': trcfg.channel_selector,
+            'sample_rate': sample_rate,
+        }
+
+        if trcfg.augmentor:
+            ds_config['augmentor'] = trcfg.augmentor
+
+        return ds_config
+
+    @abstractmethod
+    def _transcribe_input_manifest_processing(
+        self, audio_files: List[str], temp_dir: str, trcfg: TranscribeConfig
+    ) -> Dict[str, Any]:
+        pass
+
+    @abstractmethod
+    def _setup_transcribe_dataloader(self, config: Dict) -> DataLoader:
+        pass
+
+    @abstractmethod
+    def _transcribe_forward(self, batch: Any, trcfg: TranscribeConfig):
+        pass
+
+    @abstractmethod
+    def _transcribe_output_processing(self, outputs, trcfg: TranscribeConfig) -> GenericTranscriptionType:
+        pass
+
+    def _transcribe_on_end(self, trcfg: TranscribeConfig):
+        # set mode back to its original value
+        self.train(mode=trcfg._internal.training_mode)
+
+        if hasattr(self, 'preprocessor'):
+            if hasattr(self.preprocessor, 'featurizer') and hasattr(self.preprocessor.featurizer, 'dither'):
+                self.preprocessor.featurizer.dither = trcfg._internal.dither_value
+
+            if hasattr(self.preprocessor, 'featurizer') and hasattr(self.preprocessor.featurizer, 'pad_to'):
+                self.preprocessor.featurizer.pad_to = trcfg._internal.pad_to_value
+
+        if trcfg._internal.logging_level is not None:
+            logging.set_verbosity(trcfg._internal.logging_level)
+
+    def _setup_transcribe_tensor_dataloader(self, config: Dict, trcfg: TranscribeConfig) -> DataLoader:
+        dataset = TranscriptionTensorDataset(config)
+
+        # Import collate function here to avoid circular imports
+        from nemo.collections.asr.data.audio_to_text import _speech_collate_fn
```

Comment:
I am not sure if all models use same `_speech_collate_fn` name

Reply:
Right I know, but then the subclass needs to implement this function. It's fine for now, we'll have time to test the test of the models